.github/workflows/test.yaml

```diff
@@ -229,33 +306,55 @@ jobs:
         # ${{ github.workspace }}/bin/govendor remove github.com/snapcore/secboot
         # ${{ github.workspace }}/bin/govendor remove +unused
         SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=nosecboot ./run-checks --unit
+
     - name: Test Go (faultinject)
-      if: steps.cached-results.outputs.already-ran != 'true'
+      if: ${{ matrix.unit-scenario == 'faultinject' }}
       run: |
         cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
         SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=faultinject ./run-checks --unit
+
+    - name: Upload the coverage results
+      if: ${{ matrix.gochannel != 'latest/stable' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: coverage-files
+        path: "${{ github.workspace }}/src/github.com/snapcore/snapd/.coverage/coverage*.cov"
+
+  code-coverage:
+    needs: [unit-tests]
+    runs-on: ubuntu-20.04
+    env:
+      GOPATH: ${{ github.workspace }}
+      # Set PATH to ignore the load of magic binaries from /usr/local/bin And
+      # to use the go snap automatically. Note that we install go from the
+      # snap in a step below. Without this we get the GitHub-controlled latest
+      # version of go.
+      PATH: /snap/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:${{ github.workspace }}/bin
+      GOROOT: ""
+      GITHUB_PULL_REQUEST: ${{ github.event.number }}
+    steps:
+    - name: Download the coverage files
+      uses: actions/download-artifact@v3
+      with:
+        name: coverage-files
```

Comment:
Since the artifact name is always `coverage-files` regardless of the scenario that runs, I was wondering if this will properly download 4 artifacts and merge the contents in a single directory. it appears that it does:
```
Starting download for coverage-files
Directory structure has been setup for the artifact
Total number of files that will be downloaded: 4
Artifact coverage-files was downloaded to /home/runner/work/snapd/snapd/.coverage
```

Reply:
added ls to list the files used by covertool
Thanks for the suggestion